### PR TITLE
[CHORE] Migrate to Watermelon

### PR DESCRIPTION
**Pull request by diegolmello**
@RocketChat/ReactNative

- [X] Configure (iOS and Android)
- [X] Load rooms
- [X] Load room
- [X] Update rooms (socket)
- [X] Sync rooms
- [X] Update room (socket)
- [X] Sync room
- [X] Load more messages for room
- [x] Multiple servers
- [x] Share extension
- [X] Message actions (reply, edit, pin, star, etc)
- [x] Error actions (try again, delete message)
- [x] Emoji picker
- [x] User status (active users)
- [x] Permissions
- [x] Settings
- [x] Roles
- [x] Slash commands
- [x] Threads
- [X] Discussions
- [x] Uploads
- [x] User typing
- [X] Clear database on logout
- [x] Fav, hide, mark as unread room
- [X] Sort rooms
- [X] Group rooms
- [X] Search rooms
- [x] Save user
- [x] Unread count on back button
- [x] Save Notifications
- [x] Receive Notifications
- [x] Deep linking
- [x] Edit room
- [x] Draft
- [x] Work offline
- [x] Muted
- [x] NewMessage
- [x] Add user
- [x] Create channel
- [x] Test OAuth
- [x] Test sync credentials with iOS native
- [x] Performance
- [x] E2E tests
- [x] Remove Realm completely
- [x] Test again (:

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -6,10 +6,7 @@ buildscript {
         compileSdkVersion = 28
         targetSdkVersion = 28
         glideVersion = "4.9.0"
-        // googlePlayServicesVersion = "17.0.0"
-        // supportLibVersion = "1.0.2"
-        // mediaCompatVersion = '1.0.1'
-        // supportV4Version = '1.0.0'
+        kotlin_version = '1.3.21'
```
Comment (filipedelimabrito) [inline android/build.gradle]: Where is Kotlin being used here? Try using a newer version (aka 1.3.50 at the time of this comment).

Comment (filipedelimabrito) [reply android/build.gradle]: Shouldn't better to declare it like `kotlinVersion` instead? This way we keep the same naming "convection" being used here to declare the libraries version.

File: app/containers/MessageActions.js [hunk 2]
```diff
@@ -159,10 +165,11 @@ class MessageActions extends React.Component {
 		}
 	}
 
-	isOwn = props => props.actionMessage.u && props.actionMessage.u._id === props.user.id;
+	isOwn = props => props.message.u && props.message.u._id === props.user.id;
 
 	isRoomReadOnly = () => {
 		const { room } = this.props;
+		// TODO: We need to get -> is ReadOnly && !isOwner
```
Comment (djorkaeffalexandre) [inline app/containers/MessageActions.js]: You can use it: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/utils/room.js#L9, probably it is better.

File: app/containers/MessageBox/index.js [hunk 3]
```diff
@@ -324,114 +336,49 @@ class MessageBox extends Component {
...
-				});
-			}
-		} catch (e) {
-			console.warn('spotlight canceled');
-		} finally {
-			delete this.oldPromise;
-			this.users = database.objects('users').filtered('username CONTAINS[c] $0', keyword).slice(0, MENTIONS_COUNT_TO_DISPLAY);
-			this.getFixedMentions(keyword);
-			this.setState({ mentions: this.users });
+			result = [{ _id: -1, username: 'all' }, ...result];
```
Comment (djorkaeffalexandre) [inline app/containers/MessageBox/index.js]: We will use `all` instead `here` always?

File: app/views/RoomInfoEditView/index.js [hunk 4]
```diff
@@ -102,15 +99,16 @@ class RoomInfoEditView extends React.Component {
 			const observable = await db.collections
 				.get('subscriptions')
 				.query(Q.where('rid', rid))
-				.observeWithColumns(['_updated_at']);
+				.observeWithColumns(['archived', '_updated_at']);
```
Comment (diegolmello) [inline app/views/RoomInfoEditView/index.js]: `room_updated_at` instead of `_updated_at`.

File: app/containers/message/Broadcast.js [hunk 5]
```diff
@@ -11,6 +11,10 @@ import I18n from '../../i18n';
 const Broadcast = React.memo(({
 	author, user, broadcast, replyBroadcast
 }) => {
+	// FIXME: do we need this?
+	if (!author) {
```
Comment (diegolmello) [inline app/containers/message/Broadcast.js]: Do we still need this?

File: app/views/RoomInfoEditView/index.js [hunk 6]
```diff
@@ -66,57 +65,72 @@ class RoomInfoEditView extends React.Component {
...
+		}
+		try {
+			const db = database.active;
+			const observable = await db.collections
+				.get('subscriptions')
+				.query(Q.where('rid', rid))
+				.observeWithColumns(['archived', '_updated_at']);
+
+			this.querySubscription = observable.subscribe((data) => {
+				const [room] = data;
```
Comment (diegolmello) [inline app/views/RoomInfoEditView/index.js]: Nope

File: app/views/RoomView/List.js [hunk 7]
```diff
@@ -1,87 +1,144 @@
...
+		this.messagesSubscription = this.messagesObservable
+			.subscribe((data) => {
+				this.interaction = InteractionManager.runAfterInteractions(() => {
+					if (tmid) {
+						data = [this.thread, ...data];
+					}
+					const messages = orderBy(data, ['ts'], ['desc']);
+					if (this.mounted) {
+						LayoutAnimation.easeInEaseOut();
+						this.setState({ messages });
```
Comment (diegolmello) [inline app/views/RoomView/List.js]: Create something like RoomsListView's allChats state to better update.

File: app/views/RoomView/index.js [hunk 8]
```diff
@@ -449,17 +535,45 @@ class RoomView extends React.Component {
 			});
 		} catch (e) {
 			log(e);
+			console.log(e);
```
Comment (diegolmello) [inline app/views/RoomView/index.js]: Remove

File: app/views/RoomView/index.js [hunk 9]
```diff
@@ -471,6 +585,7 @@ class RoomView extends React.Component {
 			EventEmitter.emit(LISTENER, { message: isFollowingThread ? 'Unfollowed thread' : 'Following thread' });
 		} catch (e) {
 			log(e);
+			console.log(e);
```
Comment (diegolmello) [inline app/views/RoomView/index.js]: Remove

File: app/views/RoomsListView/ServerDropdown.js [hunk 10]
```diff
@@ -34,15 +34,21 @@ class ServerDropdown extends Component {
...
 		this.animatedValue = new Animated.Value(0);
-		safeAddListener(this.servers, this.updateState);
 	}
 
-	componentDidMount() {
+	async componentDidMount() {
+		const serversDB = database.servers;
+		const observable = await serversDB.collections
+			.get('servers')
+			.query()
```
Comment (diegolmello) [inline app/views/RoomsListView/ServerDropdown.js]: Does it work without `query().fetch()`?

**Pull request merged**